### PR TITLE
Extend search

### PR DESCRIPTION
**Pull request by Kamuso**
Can now search by file type, file name, md5 hash, sha1 hash. Also with tag completion.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -51,10 +51,12 @@ var (
   followed by one of these comparison operators:
     >, <, =, >=, <=
   and a positive integer.
+  There is also the type system tag to search by file type.
   Examples:
     hydron search system:width>1920 system:height>1080 artist:null
     hydron search system:tag_count=0 order:random
-    hydron search 'red_scarf -bed system:size<10485760'`,
+    hydron search 'red_scarf -bed system:size<10485760'
+	hydron search system:type=gif`,
```
Comment (bakape) [inline main.go]: Indentation is off.

File: templates/image.qtpl [hunk 2]
```diff
@@ -22,7 +24,7 @@
 			<section id="tags">
 				{% if img.Name != "" %}
 					<span class="image-name">
-						<a href="/search?q={%s img.Name %}" title="Search for name">
+						<a href='/search?q={%s url.QueryEscape("name:") + img.Name %}' title="Search for name">
```
Comment (bakape) [inline templates/image.qtpl]: Why `'`?

**Review by bakape — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (Kamuso) [reply main.go]: Didn't notice the other lines used spaces instead of tab.

File: templates/image.qtpl (hunk 2 above)
Comment (Kamuso) [reply templates/image.qtpl]: IDE syntax highlighting gets a bit ugly in .qtpl with nested strings, in my opinion. Changed to be consistent.

**Review by bakape — APPROVED** [review]

**Comment by bakape** [comment]
Thanks!

**Pull request merged**